gnovm/pkg/gnolang/preprocess.go

```diff
@@ -1582,8 +1582,16 @@ func Preprocess(store Store, ctx BlockNode, n Node) Node {
 							lhs0 := n.Lhs[0].(*NameExpr).Name
 							lhs1 := n.Lhs[1].(*NameExpr).Name
 
-							dt := evalStaticTypeOf(store, last, cx.X)
-							mt := baseOf(dt).(*MapType)
+							var mt *MapType
+							st := evalStaticTypeOf(store, last, cx.X)
+							if dt, ok := st.(*DeclaredType); ok {
+								if mt, ok = dt.Base.(*MapType); !ok {
+									s := fmt.Sprintf("the base of the %s is invalid, it should be MapType", dt.Name)
+									panic(s)
+								}
+							} else if mt, ok = st.(*MapType); !ok {
+								panic("invalid index expression on MapType")
+							}
```

Comment:
```suggestion
							var mt *MapType
							dt := evalStaticTypeOf(store, last, cx.X)
							mt, ok := baseOf(dt).(*MapType)
							if !ok {
								panic(fmt.Sprintf("invalid index expression on %T", dt))
							}
```

baseOf does the exact thing we're doing here, no? I don't see much advantage to expanding the function as you did.

Reply:
I agree with Morgan.

Reply:
Agree, Done.